cellarium/ml/core/module.py

```diff
@@ -416,3 +416,23 @@ def on_save_checkpoint(self, checkpoint: dict[str, Any]) -> None:
                 checkpoint["loops"]["fit_loop"]["epoch_progress"]["total"]["completed"] += 1
                 checkpoint["loops"]["fit_loop"]["epoch_progress"]["current"]["completed"] += 1
                 checkpoint["CellariumAnnDataDataModule"]["epoch"] += 1
+
+    def on_train_end(self) -> None:
```

Comment:
I do not think this is necessary.  See the comment on the `on_end()` function in nmf.py